src/hooks/useAuthCheck.ts

```diff
@@ -0,0 +1,209 @@
+import { logger } from "config/logger";
+import { AUTH_STATUS } from "config/types";
+import { useAuthenticationStore } from "ducks/auth";
+import { useEffect, useRef, useState, useCallback } from "react";
+import {
+  AppState,
+  AppStateStatus,
+  InteractionManager,
+  PanResponder,
+  PanResponderInstance,
+} from "react-native";
+
+const BACKGROUND_CHECK_INTERVAL = 60000; // Check every minute when app is in background
+const FOREGROUND_CHECK_INTERVAL = 10000; // Check every 10 seconds when app is in foreground
+const ACTIVE_CHECK_INTERVAL = 5000; // Check every 5 seconds when user is actively using the app
+
+/**
+ * Hook that periodically checks authentication status and redirects to lock screen when needed
+ * Accelerates checks when the user is actively using the app
+ */
+const useAuthCheck = () => {
+  const { getAuthStatus, authStatus, navigateToLockScreen } =
+    useAuthenticationStore();
+  const [isActive, setIsActive] = useState(true);
+  const appState = useRef(AppState.currentState);
+  const checkIntervalRef = useRef<NodeJS.Timeout | null>(null);
+  const lastInteractionRef = useRef(Date.now());
+  const lastCheckRef = useRef(Date.now());
+  const panResponderRef = useRef<PanResponderInstance | null>(null);
+
+  // Check auth status based on current state (background, foreground, active)
+  const checkAuth = useCallback(async () => {
+    try {
+      const now = Date.now();
+      // If it's been too soon since last check, skip
+      if (now - lastCheckRef.current < 1000) {
+        return;
+      }
+
+      // Don't perform auth checks too frequently while on lock screen
+      if (authStatus === AUTH_STATUS.HASH_KEY_EXPIRED) {
+        const skipChecksWhenExpired = true;
+        if (skipChecksWhenExpired) {
+          // Skip frequent checks when already in expired state
+          return;
+        }
+      }
```

Comment:
this could be simplified to this:
```suggestion
if (authStatus === AUTH_STATUS.HASH_KEY_EXPIRED) {
  return;
}
```